src/Confluent.Kafka/Consumer.cs

```diff
@@ -133,30 +133,30 @@ private void LogCallback(IntPtr rk, SyslogLevel level, string fac, string buf)
 
             if (err == ErrorCode.Local_AssignPartitions)
             {
-                if (partitionsAssignedHandler != null)
+                if (rebalanceHandler != null)
                 {
                     assignCallCount = 0;
```

Comment:
needs locking

Reply:
yeah, good point. same sentiment as before (wouldn't like to be reading who-ever's code made this necessary!), but it's a bigger deal here.